src/il_ros_hsr/nets/train_action_predictor.py

```diff
@@ -88,7 +88,8 @@ def _save_images(imgs_t, imgs_tp1, labels_pos, labels_ang, out_pos,
  
         # Computing 'raw' L2, well for the (224,224) input image ...
         #L2_pix = np.linalg.norm(targ_pos_ing - pred_pos_int)
-        L2_pix = 0.0 # will do later
+        L2_pix = np.linalg.norm(np.array(targ_pos_int) - np.array(pred_pos_int))
+        L2_pix += (100 if targ_ang != pred_ang else 0) # penalize wrong angle heavily
```

Comment:
The angle should not be used for the L2 error. The angle and direction are different signals and the best way is to report error metrics separately.

What would be interesting, though, is seeing how often the failure cases happen. That is, when angle is x, what is the distribution of other errors? It would be a bad sign if we get errors in the 180 deg direction, but the 90 deg and 270 (or -90 deg) are perhaps less bad.

Reply:
Changed it. Plotting script just shows L2 pixel loss for the grasp point in the bar chart, but it also prints the number of angle differences (right now, 5/199) and the magnitude of difference ([1, 3, 3, 3, 3] - so we don't have any 180 degree errors, just 90 or -90 ones).